Cleanup multiple modules: if user is in one study site and/or in "DCC": Redmine 13485

MounaSafiHarab opened the pull request:
Cleanup the code to use the isInOneStudySite() in the User class that was introduced in #2996 for data team helper. In this series of changes, we check if the user is in at least one study site and/or at the "DCC" (or centerID=1).

In reference to Redmine 13485, and related to redmine 12946;

MounaSafiHarab commented on modules/new_profile/php/NDB_Form_new_profile.class.inc:
```diff
@@ -34,14 +34,9 @@ class NDB_Form_New_Profile extends NDB_Form
-        $oneIsStudySite = in_array("1", $isStudySite);
-        $oneIsDCCSite   = in_array("DCC", $isDCCSite);
-        if ($oneIsStudySite or $oneIsDCCSite) {
+        $userInDCCSite   = in_array("1", $site_arr);
```
Here I check for CenterID rather than the hardcoded "DCC" which is not the CenterName anyway on any new Loris install (it is Data Coordinating center not DCC as DCC is the Alias).

driusan commented:
I'm fairly sure Redmine 143486 is a typo..

MounaSafiHarab commented:
> I'm fairly sure Redmine 143486 is a typo..

it is indeed!

driusan commented on php/libraries/User.class.inc:
```diff
@@ -317,6 +317,21 @@ class User extends UserPermissions
+     *
+     * @return boolean
+     */
+    function isInOneStudySite()
```
I'm not a fan of the prefix "isInOne" because it sounds like it means must have exactly one. What about "hasStudySite"?

driusan commented on php/libraries/User.class.inc:
```diff
@@ -317,6 +317,21 @@ class User extends UserPermissions
+        $site_arr         = $this->userInfo['CenterIDs'];
+        foreach ($site_arr as $key=>$val) {
+            $site[$key]        = & Site::singleton($val);
+            $isStudySite[$key] = $site[$key]->isStudySite();
```
can't you just have an early return here and then return false if nothing is found in the loop?

driusan commented on modules/candidate_list/php/candidate_list.class.inc:
```diff
@@ -34,17 +34,12 @@ class Candidate_List extends \NDB_Menu_Filter
         return (
             $user->hasPermission('access_all_profiles')
-            || ($oneIsStudySite && $user->hasPermission('data_entry'))
+            || ($userInStudySite && $user->hasPermission('data_entry'))
```
This seems to be the only place that the variable is used. Why not just inline the function call and get rid of the variable?

(Same with in all the other files)

driusan commented on php/libraries/User.class.inc:
```diff
@@ -317,6 +317,24 @@ class User extends UserPermissions
+            $isStudySite[$key] = $site[$key]->isStudySite();
+            if (in_array("1", $isStudySite)) {
+                return true;
+            }
```
can't this whole loop just be:

```
foreach ($site_arr as $key => $sitename) {
    $site = Site::singleton($sitename);
    if ($site->isStudySite()) {
        return true;
    }
}
```

I don't see the point of creating the arrays which aren't really used.

MounaSafiHarab replied:
me neither! total oversight (just simply copied and pasted the function that initially was in the codebase without cleaning it). Fixed now. Thanks.

MounaSafiHarab commented:
@kongtiaowang 
I removed the passed manual test simply because I incorporated Dave's feedback.
Do you think you can do another quick round of testing please?

MounaSafiHarab commented:
@kongtiaowang 
I incorporated Dave's feedback, so another round of manual testing is needed please.
